evalml/automl/automl_algorithm/default_algorithm.py

```diff
@@ -484,35 +489,40 @@ def _transform_parameters(self, pipeline, proposed_parameters):
         return parameters
 
     def _make_split_pipeline(self, estimator, pipeline_name=None):
-        numeric_pipeline_parameters = {
-            "Select Columns Transformer": {"columns": self._selected_cols}
-        }
-        numeric_pipeline = make_pipeline(
-            self.X,
-            self.y,
-            estimator,
-            self.problem_type,
-            sampler_name=self.sampler_name,
-            parameters=numeric_pipeline_parameters,
-            extra_components=[SelectColumns],
-            extra_components_position="before_estimator",
-            use_estimator=False if self._selected_cat_cols else True,
-        )
-
         if self._selected_cat_cols:
             self._split = True
+
+            if self._X_with_cat_cols is None or self._X_without_cat_cols is None:
+                self._X_without_cat_cols = self.X.drop(self._selected_cat_cols, axis=1)
+                self._X_with_cat_cols = self.X[self._selected_cat_cols]
+
+            numeric_pipeline_parameters = {
```

Comment:
@jeremyliweishih I think numeric vs categorical is not the right distinction. It's categorical vs non-categorical. I think right now we're dropping boolean features (and anything that is not numeric or categorical, e.g. natural language)

```python
from evalml.automl import AutoMLSearch
import pandas as pd
import woodwork as ww
from evalml.demos import load_fraud

X, y = load_fraud(1000)

automl = AutoMLSearch(X, y, "binary", verbose=True, _automl_algorithm="default",
                      max_batches=3)
automl.search()

assert X.ww['customer_present'].ww.logical_type.type_string == 'boolean' 
assert 'customer_present' in automl._automl_algorithm._selected_cols
assert not 'custsomer_present' in automl.best_pipeline.transform_all_but_final(X).columns
```

Reply:
ah good catch, will fix!

Reply:
@freddyaboulton will rerun perf tests as well!